### ISSUE TITLE
Upppercase L + quote/apostrophe kerning should be improved

### ISSUE THREAD

**Issue by jmontane** [google/roboto#195]
Hi,
Uppercase L followed by apostrophe character ' are adjusted for English quote case use, but other languages use ' char for elisions. Mainly French, and other languages like Catalan, use ' to "merge" two words, and current «L'» Roboto kerning joins these words excessively.

For instance,
Le+arbre (French words for «The tree» ), become «L'arbre». And the weird effect with «L'» kerning arises.
 
Following, an screen-capture showing several Catalan elisions. Please, note lack of space on «L'agrada» compared with other elisions.

![lquote](https://cloud.githubusercontent.com/assets/339645/16407598/d0702be6-3d14-11e6-9009-d4a676dc1cc9.png)

**Comment by fitojb** [comment]
An interesting read: http://typedrawers.com/discussion/2032/french-apostrophe

**Comment by phcannesson** [comment]
Is there any update on this ?

**Comment by phcannesson** [comment]
Could we have an update on this ? " L' " is VERY common in french and Roboto is the only font I know that is merging so close the word and the uppercase L.

I can't believe nobody is pushing for a fix...

**Comment by jamesgk** [comment]
This should be trivially fixable by removing (or adjusting) the "L / singlequote" kerning rule in source:

```diff
diff --git a/src/v2/Roboto-Bold.ufo/kerning.plist b/src/v2/Roboto-Bold.ufo/kerning.plist
index 64c434973..1dbec3018 100644
--- a/src/v2/Roboto-Bold.ufo/kerning.plist
+++ b/src/v2/Roboto-Bold.ufo/kerning.plist
@@ -1882,8 +1882,6 @@
     <integer>-71</integer>
     <key>A</key>
     <integer>23</integer>
-    <key>quotesingle</key>
-    <integer>-264</integer>
     <key>space</key>
     <integer>-30</integer>
   </dict>
diff --git a/src/v2/Roboto-Regular.ufo/kerning.plist b/src/v2/Roboto-Regular.ufo/kerning.plist
index 3c0477914..23d585c0d 100644
--- a/src/v2/Roboto-Regular.ufo/kerning.plist
+++ b/src/v2/Roboto-Regular.ufo/kerning.plist
@@ -2087,8 +2087,6 @@
     <integer>-64</integer>
     <key>A</key>
     <integer>19</integer>
-    <key>quotesingle</key>
-    <integer>-288</integer>
   </dict>
   <key>K</key>
   <dict>
diff --git a/src/v2/Roboto-Thin.ufo/kerning.plist b/src/v2/Roboto-Thin.ufo/kerning.plist
index 45827bf09..c014bb677 100644
--- a/src/v2/Roboto-Thin.ufo/kerning.plist
+++ b/src/v2/Roboto-Thin.ufo/kerning.plist
@@ -1901,8 +1901,6 @@
     <integer>-65</integer>
     <key>A</key>
     <integer>19</integer>
-    <key>quotesingle</key>
-    <integer>-336</integer>
   </dict>
   <key>K</key>
   <dict>
```

But of course, it's ultimately up to the people with authority on Roboto's design. @phcannesson, submitting a pull request may help further the discussion.

**Comment by lamarqua** [comment]
Bump. Is this gonna be in the release at some point?

**Comment by davelab6** [comment]
Next release is due before EOY

**Comment by davelab6** [comment]
Ah, that release won't address this.

**Comment by mnogaret** [comment]
The L and the A touches only when separated by an apostrophe:
<img width="86" alt="roboto" src="https://user-images.githubusercontent.com/12430425/63013651-fd8bb080-be8c-11e9-9c94-6071aea16f17.png">
Capture From [Google Fonts](https://fonts.google.com/specimen/Roboto?selection.family=Roboto).
This looks weird and it hurts readability of text in French.

**Comment by phcannessondkt** [comment]
The Pull Request for this is still pending. We'll soon celebrate its two-years anniversary.

**Comment by lamarqua** [comment]
@davelab6 Will this ever be fixed?

**Comment by xperseguers** [comment]
Any news on this? Crazy that nobody is taking care of this.

**Comment by lamarqua** [comment]
We have celebrated the 4th anniversary of this bug 🎉 (a month ago, actually.)

**Comment by fitojb** [comment]
This is addressed already in the future redesign of Roboto by Type Network.

(Please refrain from posting “funny” comments on the birthdays of bugs; it’s annoying for everyone.)

**Comment by xperseguers** [comment]
Great to have finally some news then! Just wondering what it exactly means if “it’s addressed already” but... “in a FUTURE redesign [of the font]”.

I’m not the ticket’s author but I guess a link to the merged fixed/comment/whatever and this ticket may then be closed for good!

**Comment by jamesgk** [comment]
(Not to put words in other people's mouths @fitojb, but I don't think these birthday comments are meant to be "funny", I think they're meant to express frustration that this issue has remained unresolved for *years* despite being marked high priority.)

**Comment by lamarqua** [comment]
Exactly. And had I not posted this comment, we likely wouldn't have gotten an update, so it did serve a useful purpose.

**Comment by fitojb** [comment]
Maybe use Google next time and save your birthday wishes to Facebook. This is a bug tracker, not a forum.

**Comment by jamesgk** [comment]
Oh, the irony of complaining about "annoying" posts pinging a bug, and then turning around and posting such an adversarial message baiting an argument! 😂

I'm not going to engage here any further than saying this:
- There's absolutely no excuse for this bug still being unresolved. No matter what the format of Roboto's current sources is, this should be extremely easy to fix.
- The fact that this bug is still unresolved is a failure of the bug tracker. Pinging the bug report is a perfectly valid way to try to remedy that failure.
- Searching "roboto french L apostrophe" on Google leads straight to this bug report.

**Comment by m4rc1e** [comment]
Thought I'd just give an update.

We're a few weeks away from pushing Roboto VF. I need to make a new release so I'm going to include a fix for this.

Can anyone confirm that the bad kerning is only happening to the "L" + "singlequote"? If so, I'll happily remove these kerns.

**Comment by xperseguers** [comment]
I recompiled the font using the provided patch and I am using it on a French only website. Never found any other problem with and would have loved not having to set up a whole Ubuntu VM just for that but life is life :-)

**Comment by phcannessondkt** [comment]
Has it been released to production ? On my Android device or any Google service, I still see the issue.

**Comment by phcannesson** [comment]
Is there any update on this ?
